No rule to make target 'build/SDL.lo'

Hi,

I have been trying to compile Oni2 in ArchLinux but I keep getting the following error:

`make: *** No rule to make target 'build/SDL.lo', needed by 'build/libSDL2.la'.  Stop.`

[Here](https://pastebin.com/veSf5UDT) is the output of `esy build` which includes the output of ./configure and make.

I have everything listed [here](https://github.com/revery-ui/revery/wiki/Building-&-Installing) installed and I'm suspecting it is not a missing dependency because I'm able to compile it using cmake (the official Arch package is also built using [cmake](https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/sdl2))

I'm pretty sure I'll figure it out eventually but I appreciate if anyone has any leads

--- a/build-scripts/esybuild.sh
+++ b/build-scripts/esybuild.sh
@@ -1,0 +1,72 @@
+#!/bin/bash
+
+# =====================================
+# Globals
+# =====================================
+
+# shellcheck disable=SC2154
+readonly INSTALL_PREFIX=$cur__install
+readonly SOURCE_PATH=$(pwd)
+readonly BUILD_PATH="${SOURCE_PATH}/esy_build"
+
+# =====================================
+# Utils
+# =====================================
+
+# SDL2 build instructions specifies that we should use
+# a separate build directory instead of building in the
+# base of the source tree.
+# see: https://wiki.libsdl.org/Installation
+execute_inside_build_dir() {
+  [ ! -d "$BUILD_PATH" ] && mkdir "$BUILD_PATH"
+
+  (cd "$BUILD_PATH" && "$@")
+}
+
+make_and_install() {
+  make && make install
+}
+
+# =====================================
+# Setup functions
+# =====================================
+
+setup_macos() {
+  "${SOURCE_PATH}"/configure \
+    --enable-shared=no \
+    --enable-static=yes \
+    --disable-video-x11 \
+    --prefix="$INSTALL_PREFIX"
+}
+
+setup_windows() {
+  SHELL=bash CC=x86_64-w64-mingw32-gcc WINDRES=x86_64-w64-mingw32-windres SDL_THREAD_PTHREAD=1 "${SOURCE_PATH}"/configure \
+    --prefix="$INSTALL_PREFIX" \
+    --enable-pthreads
+}
+
+setup_linux() {
+  CONFIG_SHELL=bash CFLAGS="-fPIC -fcommon" "${SOURCE_PATH}"/configure --prefix="$INSTALL_PREFIX"
+}
+
+while [ "$1" != "" ]; do
+  case $1 in
+    --os=darwin)
+      execute_inside_build_dir setup_macos
+      break
+      ;;
+    --os=linux)
+      execute_inside_build_dir setup_linux
+      break
+      ;;
+    --os=windows)
+      execute_inside_build_dir setup_windows
+      break
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
+
+execute_inside_build_dir make_and_install

--- a/package.json
+++ b/package.json
@@ -5,27 +5,27 @@
   "license": "MIT",
   "esy": {
     "build": [
-        ["bash", "-c", "#{os == 'windows' ? 'SHELL=bash CC=x86_64-w64-mingw32-gcc WINDRES=x86_64-w64-mingw32-windres SDL_THREAD_PTHREAD=1 ./configure --prefix=$cur__install --enable-pthreads': 'echo'}"],
-        ["bash", "-c", "#{os == 'darwin' ? './configure --enable-shared=no --enable-static=yes --disable-video-x11 --prefix=$cur__install' : 'echo'}"],
-        ["bash", "-c", "#{os == 'linux' ? 'CFLAGS=-fPIC ./configure --prefix=$cur__install' : 'echo'}"],
-        ["make"],
-        ["make", "install"]
+      [
+        "bash",
+        "-c",
+        "./build-scripts/esybuild.sh --os=#{os}"
+      ]
     ],
-    "buildsInSource":true,
+    "buildsInSource": true,
     "exportedEnv": {
-        "SDL2_INCLUDE_PATH": {
-            "val": "#{self.install / 'include'}",
-            "scope": "global"
-        },
-        "SDL2_LIB_PATH": {
-            "val": "#{self.lib}",
-            "scope": "global"
-        },
-        "LD_LIBRARY_PATH": {
-            "val": "#{self.lib : $LD_LIBRARY_PATH}",
-            "scope": "global"
-        }
+      "SDL2_INCLUDE_PATH": {
+        "val": "#{self.install / 'include'}",
+        "scope": "global"
+      },
+      "SDL2_LIB_PATH": {
+        "val": "#{self.lib}",
+        "scope": "global"
+      },
+      "LD_LIBRARY_PATH": {
+        "val": "#{self.lib : $LD_LIBRARY_PATH}",
+        "scope": "global"
+      }
     }
   },
-  "dependencies": { }
+  "dependencies": {}
 }

